### ISSUE TITLE
HTTP "fetch a file from a URL" needs to follow redirect

### ISSUE THREAD

**Issue by dagolden** [andk/pause#126]
Per email, I observed that PAUSE downloads a corrupt tarball from a Dropbox share URL.

When testing that URL manually, a redirect is required to retrieve the tarball.

**Comment by wolfsage** [comment]
PAUSE appears to handle redirects fine.

This worked: http://hiddenrealms.org/redirect-me/CPAN-Test-Dummy-alh-test-0.09.tar.gz (which redirects to http://hiddenrealms.org/CPAN-Test-Dummy-alh-test-0.09.tar.gz)

What I noticed when I tried this from DropBox (https://www.dropbox.com/s/bo1g3wbjex3zg6l/CPAN-Test-Dummy-alh-test-0.09.tar.gz?dl=1) is that the file was entered in as:

CPAN-Test-Dummy-alh-test-0.09.tar.gzdl1 (the query parameters were appended to the filename).
The md5sum of the file was still correct though.

Then, mldiswatch attempted to pick up the file, but I think it said "gzdl1 is not a valid extension, go away."

I'm not sure the logic of handling url downloads with query parameters is working right.
